src/lib.rs

```diff
@@ -4139,11 +3967,7 @@ fn get_m_opcode_and_encoding(tag: u8, word: &BitSlice<Lsb0, u8>) -> (Opcode, Ope
             ];
 
             let index = ((word[36] as u8) << 1) + (word[27] as u8);
-            if let Some(op_table) = TABLE4_28[index as usize] {
-                op_table[word[30..36].load::<u8>() as usize]
-            } else {
-                (Purple, None)
-            }
+            TABLE4_28[index as usize].map_or((Purple, None), |op_table| op_table[word[30..36].load::<u8>() as usize])
```

Comment:
i generally find `map_or` less legible, do you have a preference one way or the other? i don't find it so bad as to ask you to revert the `map_or` clippies, but if you were on the fence about them, you're welcome to just add the right anti-clippy incantation for these.

Reply:
There were a bunch of anti-clippy lints I added here; for example it wanted to more aggressively unify some match arms which had TODO comments in them, and I've tried to err in favour of not doing that.

`map_or` is something which can situationally improve legibility, or equally make it worse, so yeah I'm generally on the fence there. I thought it helped when I was writing that snippet for checking if there was a stop bit before alloc in the second slot, however, and I also think it's a bit better here too.